### ISSUE TITLE
bug(datepicker): It should announce the month name instead of number, when reading the focused date

### ISSUE THREAD

**Issue by zarend** [angular/components#23479]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the "basic datepicker" demo
3. Activate "Open calendar" button from any section, say, "Single date selection"
4. Focus on any date value
 
#### Expected Behavior
It should announce the month name instead of month number for better understanding by the screen readers

#### Actual Behavior
It is announcing the month number instead of name. For example, it is announcing as "Thursday 08 5, Column 5 of 7"

#### Environment
OS| Browser | Screen reader
macOS Catalina | Google Chrome 92.0.4515.131 | VoiceOver

##### Additional Notes
This is copied from an [internal bug report](http://b/195600348) found during a11y testing.

**Comment by zarend** [comment]
I tried again, and I still can't reproduce this. I couldn't find where a numeric month would be coming from each cell on the calendar has an aria label with the full date (e.g. aria-label="February 1, 2022")

MacOS 12.2 (21D49), Chrome Version 97.0.4692.99 (Official Build) (x86_64), VoiceOver
